### ISSUE TITLE
Support for Secrets Manager

### ISSUE THREAD

**Issue by holmesjr** [buildkite/elastic-ci-stack-for-aws#460]
Rather than writing secrets into a bucket with the default KMS key, it would be great to have support for [Secrets Manager](https://aws.amazon.com/secrets-manager/).

**Comment by holmesjr** [comment]
Perhaps it can be handled here: https://github.com/buildkite/elastic-ci-stack-s3-secrets-hooks/issues/21

**Comment by holmesjr** [comment]
The files are:

- [bootstrap](https://gist.github.com/holmesjr/8c7c4009bdc6e91ce452024d7dd7baaa)
- [env](https://gist.github.com/holmesjr/4a48ed5bff90799511a5482d173994f7)
- [creds helper](https://gist.github.com/holmesjr/062f2870bfc6d189a96e4b885c5b4685)

**Comment by lox** [comment]
Did you do any thinking about what format / structure of key you'd expect in secrets manager?

**Comment by holmesjr** [comment]
Ah yeah, that would have been useful info. I currently have this:

```
{
  "Cred": "https://USER:TOKEN@github.MYORG.com"
}
```

**Comment by holmesjr** [comment]
So it's just the git creds line as per the S3 plugin, under a "Cred" key.

**Comment by lox** [comment]
@holmesjr I spent a bit of time today expanding on those gists and wrangled it into https://github.com/buildkite/elastic-ci-stack-secrets-manager-hooks. 

The env support isn't implemented yet, and I rolled back the "use a docker image" bit. Interested in your feels @holmesjr.

**Comment by holmesjr** [comment]
You'll probably need to install pip and update the CLI that eay.

**Comment by zsims** [comment]
> Secrets get a randomised string at the end of their ARN. Any ideas are welcome.

You can still find them by secret key id (without the random parts). The random suffix of the ARN is only important for I AM resources.

FWIW we use AWS SM via a Buildkite plugin: https://github.com/seek-oss/aws-sm-buildkite-plugin (which is a different use-case to this issue)

**Comment by lox** [comment]
> Secrets get a randomised string at the end of their ARN. Any ideas are welcome.

I didn't run into that issue, I wonder what I did differently 🤔

**Comment by holmesjr** [comment]
Did you perhaps use the id rather than the ARN?

**Comment by lox** [comment]
Ah, yes!

**Comment by holmesjr** [comment]
That will work for anyone with secrets in the same account, but not cross-account. Maybe it's not a big deal and we can use a secret naming convention.

**Comment by lox** [comment]
Yeah, secrets in other accounts is something that is harder to do with this model. I think we'd probably have to do something like store a map of secret keys in a known key on the local account, e.g `elastic-ci-stack/mapping` with something like:

```json
{ 
  "elastic-ci-stack/global": "arn:aws:secretsmanager:ap-southeast-2:xxxxx:secret:buildkite-github-creds-xxxxx" 
}
```

That way you could map certain keys onto other accounts and keep the inference model, which I kind of like.
